custom_components/tahoma/cover.py

```diff
@@ -95,38 +99,31 @@ def update(self):
 
         self.controller.get_states([self.tahoma_device])
 
-        # Set current position.
+        self.update_postion()
+        self.update_tilt_position()
+        self.update_lock()
+
+    def update_position(self):
+
         # Home Assistant: 0 is closed, 100 is fully open.
         # core:ClosureState: 100 is closed, 0 is fully open.
 
+        states = self.tahoma_device.active_states
```

Comment:
Good one, I like this! Would be good to have this in all components, makes it more readable.

Reply:
Yep, it something I like to do when you repeat a lot a long path.